### ISSUE TITLE
80 bit floating point

### ISSUE THREAD

**Issue by dmrvb** [twinbasic/twinbasic#1640]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by WaynePhillipsEA** [comment]
What's the use case for this?  I take it you'd like full built-in datatype support for this, much like some C compilers support the "long double" type which maps to the x86 80-bit FPU type, then you can declare variables "Dim myValue As LongDouble" or some such.   It would also need some consideration on things constants, type suffixes, and conversions/coersions amongst other things.

**Comment by fafalone** [comment]
Hopefully adding unsigned type support is still on the post-v1 horizon, might not be too far out of the way to add more data types when that's being done. But it may cause some confusion... with Windows programming in MSVC, 'long double'/'_LONGDOUBLE' are 64bit types. The 80bit one is _LDOUBLE, implemented as `unsigned char[10]`. Then there's `_LDBL12` that's 144bit.

**Comment by WaynePhillipsEA** [comment]
Yeah some C compilers are different.  In Clang/GCC `long double` represents the 80-bit FPU type, IIRC.

**Comment by dmrvb** [comment]
> Hopefully **adding unsigned type support** is still on the post-v1 horizon, might not be too far out of the way to add more data types when that's being done. But it may cause some confusion... with Windows programming in MSVC, `long double`/`_LONGDOUBLE` are 64bit types. The 80bit one is `_LDOUBLE`, implemented as `unsigned char[10]`. Then there's `_LDBL12` that's 96bit.

Yes please, I would also really like unsigned integers (of all bit lengths). I had a quick go at writing something to read an 80bit float yesterday and concluded it would be difficult without unsigned integers>

**Comment by Kr00l** [comment]
There is also this Int128 and UInt128 (as of .Net 7)

I don't know any usecase but would that be difficult to include in tB ?
I guess this would be restricted to Win64 compile option only..
also there is no prior art for a Keyword. And I agree that `LongLongLongLong` would not be that readable.

I posted this just out of curiosity. :)

**Comment by dmrvb** [comment]
and then there is the quad real (128bits)  but Intel Hardware does not support that yet :smiley:
Wayne still has a lot to do so all these things will have to wait till the first release is well sorted.
I am a big fan of VB and generally don't like C type langages, but the long/longlong/integer names are not good, especially as in some langages they are hardware dependant. In this case I prefer to C way of doing things, int16   int32  uint32 etc (and Real32 and Real64???) so maybe we should use these in future but keep the longs and integers for backward compatability?

**Comment by fafalone** [comment]
Why not LongLongLongLong. Then we can add 256bit types and have LongLongLongLongLongLongLongLong 🤣

**Comment by FullValueRider** [comment]
I see nothing wrong with the templates 'In', 'Un' and 'Fn' where n is the number of bytes used i.e. U1 is equivalent to Byte, I2 and I4 are integer and long respectively etc.  I'd want these names to be able to be used interchangeably with current type names such that the expressions Typename(1024&) = "Long" and Typename(1024&)="I4" both return true.

**Comment by Kr00l** [comment]
Well Typename can return either Long or I4. But not both.
And it's put in stone for back-compat that it must return Long

**Comment by WaynePhillipsEA** [comment]
Perhaps we could special case string comparisons for this use case ;)

**Comment by FullValueRider** [comment]
Probably the simplest way would be to add an option to the Typename method that allows what is returned to be selected.  The default of course being VBA unless overridden by an option statement.   In the wider context changes would also have to be reflected in the range of values returned by the vartype method.
